packages/seacas/libraries/ioss/src/exodus/Ioex_BaseDatabaseIO.C

```diff
@@ -502,6 +502,24 @@ namespace Ioex {
     ex_set_max_name_length(m_exodusFilePtr, maximumNameLength);
   }
 
+  bool BaseDatabaseIO::supports_group_nl()
```

Comment:
I'm not sure this is the correct way to do this...  Basically, prior to creating the database, we want to know if the database type specified by the user supports groups -- exodus, cgns, adios, faodal, ...  Not whether a specific database file that we have already created.

So, we just have to know whether exodus was compiled with netcdf-4 capability enabled and then `EXODUS` output type would return true for supporting groups and we could then created a "grouped database".   

This would also then be a compile-time check instead of runtime and we would not need the locking and `_nl` types. 

I think checking after the database is created is too late and doesn't give the client the information that they want.  If they want to use groups and the chosen format supports groups, then they can create the database (and we make sure that netcdf-4 format is selected as we currently do).

Reply:
I get what you are saying here but it appears that we may have different interpretations of what this function is supposed to do. I think you are right that the `change-set` concept that is handled with groups in exodus is spilling out and needs to be made database-agnostic. 

Based on the usage of this function in the unit tests, my goal was to find a way to do reduction operations on the databases depending on whether an input file was written in a format that supports groups (as opposed to knowing if the currently compiled database format *can* support groups). If the input file format doesn't support groups (or `change-sets` as we are now referring to them), then we do reductions across a set of files which may be `linear` or `cyclic` , otherwise we do the reduction across the `change-sets` within the file.

I would really like to meet up to have a more in-depth discussion on these concepts and API